src/main/java/com/bfrisco/benchants/utils/Toggle.java

```diff
@@ -0,0 +1,77 @@
+package com.bfrisco.benchants.utils;
+
+import org.bukkit.*;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerInteractEvent;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import java.util.List;
+
+public class Toggle implements Listener {
+
+
+
+    @EventHandler
+    public static void activateClick(PlayerInteractEvent event) {
+
+        Player player = event.getPlayer();
+        Material coolDown = Material.JIGSAW;
+        if (!event.getAction().isRightClick()) return; //should reduce the amount of times the interactevent continues to use resources
+        ItemStack item = player.getInventory().getItemInMainHand();
+        if (!ItemInfo.isTitanTool(item)) return;
+        if (!ItemInfo.isImbued(item)) return;
+        if (player.hasCooldown(coolDown)) return;
+        player.setCooldown(coolDown,25);
+        player.sendMessage("Made it past isImbued check in toggle event");
+
+        if (!player.isSneaking()) return; //should reduce the amount of times the interactevent continues to use resources

```

Comment:
If we want to keep this, it should go right after the check for the right click.

I'm not sure if this will be needed - it may be inconvenient to not be able to toggle while shifting. Since in the vast majority of cases `ItemInfo#isTitanTool` will return false, I'm not sure we will gain any performance improvements here.